src/azul/chalice.py

```diff
@@ -93,6 +96,13 @@ def authentication_middleware(self, event, get_response):
             response = get_response(event)
         return response
 
+    def _lambda_context_middleware(self, event, get_response):
+        config.terra_client_timeout = (
+            5  # Lambda's fronted by API Gateway are too time-sensitive
+            if self.lambda_context.get_remaining_time_in_millis() < 30000 else
+            20  # In other context we can afford to be more patient
```

Comment:
```suggestion
            20  # In other contexts we can afford to be more patient
```

Reply:
We should probably also make these two values configurable via `AZUL_TERRA_TIMEOUT` set to something like `"5:20"`